framework/math/linear_solver/linear_solver.cc

```diff
@@ -79,21 +72,17 @@ LinearSolver::Setup()
 {
   if (IsSystemSet())
     return;
+
   PreSetupCallback();
 
   KSPCreate(opensn::mpi_comm, &ksp_);
 
-  // In OpenSn the PETSc version of Richardson iteration is referred to as krylov_richardson to
-  // distinguish it from classic Richardson. At this point, we need to convert from
-  // krylov_richardson to the correct PETSc algorithm name.
-  if (iterative_method_ == "krylov_richardson")
-    KSPSetType(ksp_, "richardson");
-  else
-    KSPSetType(ksp_, iterative_method_.c_str());
+  const auto petsc_iterative_method = PETScIterativeMethodName(iterative_method_);
+  KSPSetType(ksp_, petsc_iterative_method.c_str());
 
   ApplyToleranceOptions();
 
-  if (iterative_method_ == "gmres")
+  if (petsc_iterative_method == "gmres")
```

Comment:
Why not: `iterative_method_ == IterativeMethod::PETSC_GMRES`? This is IMO better than comparing strings...

Reply:
Good point. I just left the same comparison used before because I was moving quickly. Fixed.